update last released timestamp to visually connect it more to the latest version box

cheungnj opened the pull request:
This is a WIP to get quicker feedback regarding #3335 

This is what I have so far 
<img width="1154" alt="screen shot 2018-03-30 at 12 13 57 am" src="https://user-images.githubusercontent.com/6911370/38154952-bb3452b2-3429-11e8-84ef-c9179a1cf97f.png">

cheungnj commented on warehouse/filters.py:
```diff
@@ -177,3 +177,8 @@ def contains_valid_uris(items):
     return packaging.version.parse(version_str)
+
+
+def uncapitalize(text):
```
jinja has the built-in capitalize filter so I could also:

1. remove this filter
2. update [convertToReadableText](https://github.com/cheungnj/warehouse/blob/add-last-released-to-timestamp/warehouse/static/js/warehouse/utils/timeago.js#L27) to return all lowercase values
3. then invoke the capitalize filter on any existing references to humanize

cheungnj replied:
I could also use the `lower` filter. There's a lot of options...

yeraydiazdiaz replied:
`lower` would be the easiest approach, yet it reads wrong when displaying a date, i.e `Released apr 19 2018` vs `Released Apr 19 2018` (ideally `Released on April 19 2018`).

I think we should separate the Jinja side from the Javascript side. If you're up for it we could:
- Extend the function `timeago.js` to read an optional suffix from a second attribute and deal with the capitalization if present.
- Remove `Released` text and the `uncapitalized` filter so it simply displays the date as before.

cheungnj replied:
Definitely up for it but if I'm reverting my changes so the date is displayed as before, then do we really need to extend `timeago.js`?

yeraydiazdiaz replied:
I think prefixing the `Released` would be a nice touch, but it might not justify the change? thoughts @brainwane @nlhkabu?

cheungnj commented on warehouse/templates/packaging/detail.html:
```diff
@@ -91,7 +91,12 @@ <h1 class="package-header__name">
       {% endif %}
-      <p class="package-header__date">{{ humanize(release.created) }}</p>
+      <p class="package-header__date">
+        <span class="status-badge">
```
I wasn't sure if this was acceptable use of the `status-badge` class but I could also try using the `badge` class.

yeraydiazdiaz replied:
I think it's the appropriate class

brainwane commented:
@yeraydiazdiaz if you have any time to give your thoughts here, I'd appreciate it.

yeraydiazdiaz commented on warehouse/static/sass/blocks/_package-header.scss:
```diff
@@ -131,8 +135,18 @@
+      margin-top: 10px;
+
+      &:before {
+        border-color: rgba(0, 0, 0, 0.4);
```
maybe best to use the [border color defined in colours.scss](https://github.com/pypa/warehouse/blob/master/warehouse/static/sass/settings/_colours.scss#L31)

cheungnj commented:
This is what I have now. I addressed all of the comments besides extending the functionality of `timeago.js` because I wanted some clarification. 
<img width="274" alt="screen shot 2018-04-01 at 12 47 16 pm" src="https://user-images.githubusercontent.com/6911370/38176863-9424f1d0-35ab-11e8-9075-7134f3d9f6f4.png">

di commented on warehouse/templates/packaging/detail.html:
```diff
@@ -91,7 +91,12 @@ <h1 class="package-header__name">
-      <p class="package-header__date">{{ humanize(release.created) }}</p>
+      <p class="package-header__date">
+        <span class="status-badge">
+          {{ humanize(release.created) }}
```
I think prefixing this with a `Released: ` would be appropriate (not sure I totally follow the discussion in this PR about this).

cheungnj replied:
Maybe @yeraydiazdiaz meant prefix instead of suffix and the presence of that would cause the human readable date to be lowercase?

di replied:
I don't see how adding a prefix would make the date lowercase? To be clear, I'm talking about changing this line to something like:

```
Released on: {{ humanize(release.created) }}
```

cheungnj replied:
I get that but doesn't `Released on: Just now` or `Released on: About 2 hours ago` sound kinda weird? I think the discussion earlier was talking about passing the optional prefix and if it was a truthy value then we'd make the return value of `convertToReadableText` lowercase. In that case we'd need to pass the prefix only when we're converting the value for the time element in `.package-header__date`.  Maybe just doing `Released:` is the way to go.

di replied:
Yeah, I'll leave it to you to decide what sounds best, but I think the simplest approach possible here is probably ideal. `Last released:` might also work.

cheungnj replied:
That sounds like a winner to me

nlhkabu commented:
hi @cheungnj thanks for this PR.  Just to let you know that this is on my list of things to pull down and review this week, so I'll have feedback for you soon :) Thanks for your patience.

nlhkabu commented:
hi @cheungnj thanks again for this PR :)

Right now, we have some feedback asking us to reduce the whitespace in our UIs (see #1988), so I think my preference here is to keep this style simple (and taking up less space), and just add the 'Last updated' text for now.

It is likely that we will redesign this header altogether based on user feedback, so I will keep this design in mind for then.

cheungnj commented:
Here's what I have now: 
<img width="238" alt="screen shot 2018-04-28 at 1 58 36 pm" src="https://user-images.githubusercontent.com/6911370/39400816-5bcb7880-4aec-11e8-8bc5-15961c759273.png">
